qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -445,25 +501,59 @@ def _synth_natural_direction(
         wires = None
         if natural_direction in {None, True} and (coupling_map or target):
             if target is not None:
-                cmap = target.build_coupling_map(two_q_gate=decomposer2q.gate.name)
+                matching = {}
+                reverse = {}
+                # Find lowest error matching or reverse decomposer and use that
+                for index, decomposer in enumerate(decomposer2q):
+                    gate_name = getattr(decomposer, "gate_name", decomposer.gate.name)
+                    qubits_tuple = tuple(qubits)
+                    reverse_tuple = (qubits[1], qubits[0])
+                    props_dict = target[gate_name]
+                    if target.instruction_supported(gate_name, qubits_tuple):
+                        if props_dict is None or None in props_dict:
+                            error = 0.0
+                        else:
+                            error = getattr(props_dict[qubits_tuple], "error", 0.0)
+                            if error is None:
+                                error = 0.0
+                        matching[index] = error
+                    # Skip reverse check if we already have matching
+                    elif not matching and target.instruction_supported(gate_name, reverse_tuple):
+                        if props_dict is None or None in props_dict:
+                            error = 0.0
+                        else:
+                            error = getattr(props_dict[reverse_tuple], "error", 0.0)
+                            if error is None:
+                                error = 0.0
+                        reverse[index] = error
+                preferred_direction = None
+                if matching:
+                    preferred_direction = [0, 1]
+                    min_error_index = min(matching, key=matching.get)
+                    decomposer2q = decomposer2q[min_error_index]
+                elif reverse:
+                    preferred_direction = [1, 0]
+                    min_error_index = min(reverse, key=matching.get)
+                    decomposer2q = decomposer2q[min_error_index]
```

Comment:
Oops, yeah that's clearly a copy paste error, good catch. I think it is "covered" by existing tests but nothing is asserting a particular decomposer is used. The output with this is valid just not the highest basis available, I'll see if I can figure out how to assert this in a test to prevent a regression.

Reply:
I was thinking `min` would raise an exception, because `None < None` is a TypeError

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/7795/commits/10006d5bd990ae41fdd90c5acd2df3c62026a2da (it was covered and passing tests because the test cases had a list of 1 entry which didn't raise, in the fix commit I added an explicit test for this)